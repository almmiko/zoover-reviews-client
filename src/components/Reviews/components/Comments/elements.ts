import styled from 'styled-components';
import { device } from '../../../_common/Layout/media';

const LOW_RATING = 4;
const MEDIUM_RATING = 7;
const HIGH_RATING = 8;

export const Wrapper = styled.div``;
export const Header = styled.div`
  display: flex;
`;
export const RatingNumber = styled.div<{ rating: number}>`
  padding: 10px 22px;
  color: #ffff;
  font-size: 25px;
  display: inline-block;
  border-radius: 5px;
 
 background-color: ${({ rating }) => {
   
   if (rating <= LOW_RATING) {
     return'#ff3434';
   }
   
   if (rating <= MEDIUM_RATING) {
     return '#3ac135';
   }
   
   if (rating >= HIGH_RATING) {
     return '#03600d'
   }
   
  }};
  
`;
export const Title = styled.div`
  margin-left: 10px;
  display: flex;
  flex-direction: column;
  justify-content: center;
`;
export const TitleText = styled.h2`
  margin-bottom: 0;
  font-size: 16px;
`;
export const TitleSubInfo = styled.div`
  font-size: 12px;
  color: #8a8a8a;
`;

export const RatingAspects = styled.div`
  overflow: hidden; /* simple clearfix */
  margin: 10px 0;

`;
export const Aspect = styled.div`
  float: left;
  width: 25%;
  display: flex;
  justify-content: space-between;
  padding-right: 30px;
  margin-bottom: 5px;
  
  @media ${device.tablet} {
    float: inherit;
    width: 100%;
  }
`;
export const AspectRating = styled.div`
  color: #757575
`;
export const AspectDescription = styled.div`
  color: #9e9e9e
`;

export const Description = styled.div``;

export const MetaInfo = styled.div`
  margin-top: 10px;
  font-size: 12px;
  color: #8a8a8a;
`;
