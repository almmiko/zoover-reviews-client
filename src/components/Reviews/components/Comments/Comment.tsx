import React from 'react';
import { Wrapper, Header, RatingNumber, Title, TitleText, TitleSubInfo, RatingAspects, Aspect, AspectDescription, AspectRating, Description, MetaInfo } from './elements';

class Comment extends React.Component<any> {
  render() {
    return (
      <Wrapper>
        <Header>
          <RatingNumber>8</RatingNumber>
          <Title>
            <TitleText>Tittle</TitleText>
            <TitleSubInfo>user, 13-12-44</TitleSubInfo>
          </Title>
        </Header>
        <RatingAspects>
          <Aspect>
            <AspectDescription>Location</AspectDescription>
            <AspectRating>5</AspectRating>
          </Aspect>
          <Aspect>
            <AspectDescription>Location</AspectDescription>
            <AspectRating>5</AspectRating>
          </Aspect>
          <Aspect>
            <AspectDescription>Location</AspectDescription>
            <AspectRating>5</AspectRating>
          </Aspect>
          <Aspect>
            <AspectDescription>Location</AspectDescription>
            <AspectRating>5</AspectRating>
          </Aspect>
          <Aspect>
            <AspectDescription>Location</AspectDescription>
            <AspectRating>5</AspectRating>
          </Aspect>
          <Aspect>
            <AspectDescription>Location</AspectDescription>
            <AspectRating>5</AspectRating>
          </Aspect>
        </RatingAspects>
        <Description>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolorem doloremque earum incidunt quibusdam quod. Accusamus animi, aperiam est facilis iure odio provident totam voluptate. Dicta eligendi ex modi tempora tenetur!
        </Description>
        <MetaInfo>
          Traveled with family in 12.44.44
        </MetaInfo>
      </Wrapper>
    );
  }
}

export default Comment;
