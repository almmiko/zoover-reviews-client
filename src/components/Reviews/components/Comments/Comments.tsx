import { List as AntList } from 'antd';
import React from 'react';
import Comment from './Comment';
import { observer } from 'mobx-react';
import { ReviewsComments, Resource } from '../../../../typings/reviewsComments';

type Props = {
  comments: ReviewsComments,
  loaded: boolean,
};

@observer
class Comments extends React.Component<Props> {
  render() {
    const { comments, loaded }  = this.props;
    const { resources, meta } = comments;

    return(
      <AntList
        itemLayout="vertical"
        size="large"
        loading={!loaded}
        pagination={loaded && {
          onChange: page => {
            console.log(page);
          },
          pageSize: +meta.limit || 20,
          total: meta.totalItems,
        }}
        dataSource={resources}
        renderItem={(item: Resource) => (
          <AntList.Item key={item.id}>
            <Comment comment={item} />
          </AntList.Item>
        )}
      />
    );
  }
}

export default Comments;
