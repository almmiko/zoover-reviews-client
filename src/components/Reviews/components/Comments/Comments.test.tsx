import { shallow } from 'enzyme';
import Comments from './Comments';
import React from 'react';

it('renders correctly', () => {
  const tree = shallow(<Comments
    currentPage={1}
    loaded={true}
    error={false}
    comments={{
      resources: [],
      meta: {} as any,
    }}
    onPageChanged={jest.fn()}
  />);
  expect(tree).toMatchSnapshot();
});
