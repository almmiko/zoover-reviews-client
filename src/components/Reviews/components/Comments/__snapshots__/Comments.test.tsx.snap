// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Fragment>
  <List
    bordered={false}
    dataSource={Array []}
    itemLayout="vertical"
    loading={false}
    pagination={
      Object {
        "current": 1,
        "onChange": [Function],
        "pageSize": 20,
        "total": undefined,
      }
    }
    renderItem={[Function]}
    size="large"
    split={true}
  />
</Fragment>
`;
